dev/breeze/src/airflow_breeze/utils/run_utils.py

```diff
@@ -91,9 +91,9 @@ def run_command(
                 console.print("[red]========================= STDERR start ============================[/]")
                 console.print(ex.stderr)
                 console.print("[red]========================= STDERR end ==============================[/]")
-        if not check:
+        if check:
             raise
-    return None
+        return ex
```

Comment:
why do we need to return the exception?

Reply:
Because `CalledProcessorError` has the same fields as `CompletedProcess` :)

Both have:

* returncode
* stdout
* stderr

And then, we can handle it much easier (and it makes handling returned types much easier when we use `check=False`) - instead of `Optional[CompletedProcess]` we return now `Union[CalledProcessError,CompletedProcess`]. Then we can do:

```
if result.returncode != 0:
    ......
   sys.exit(result.returncode)
```

instead of something like that:

```
if not result or returncode != 0:
    ......
   sys.exit(result.returncode if result else 1)
```

This makes all the conditionals far easier.

One could ask why we need to get returncode and `check = True`. The main reason why we do it because this is a "command line tool". 

So rather than printing the user an ugly stacktrace we really want to print a nice and explanatory error message (nice with [red] marker) so we just need to exit with the right returncode rather than throw ugly exception. This is in cases where we "expect" some errors (for example we expect that "building docs" might fail - but we already handle it by printing all reasons and errors. 

We still have a number of cases when we do "check=True" - this is when we do not expect the command to fail ever (but for many reasons - misconfigured system etc it might happen). In this case we simply run `check=True` and if such command fails, ugly exception will be printed (ugly but useful for diagnostics in this case :).

Reply:
I just realised this today when I looked at the "failing but green docs" and I realized this Union is cool.

Reply:
BTW. I will add some adr about it i.e "How we are handling communication and errors in Breeze" - I have a task for it in https://github.com/apache/airflow/issues/22906